pkg/client/attach.go

```diff
@@ -22,9 +22,26 @@ const (
 )
 
 type AttachStreams struct {
-	Stdin                                   io.Reader
-	Stdout, Stderr                          io.WriteCloser
-	AttachStdin, AttachStdout, AttachStderr bool
+	// Standard input stream, can be nil.
+	Stdin *Stdin
+
+	// Standard output stream, can be nil.
+	Stdout *Stdout
+
+	// Standard error stream, can be nil.
+	Stderr *Stderr
+}
+
+type Stdin struct {
+	io.Reader
+}
+
+type Stdout struct {
+	io.WriteCloser
+}
+
+type Stderr struct {
+	io.WriteCloser
 }
```

Comment:
actually, nit, do we reallly need two structs here? we could just name it in and out and then use  the correct ones?